File: output/CodeLlama/avatar/Python/Java/atcoder_ABC132_F.java
import java.util.*;
import java.lang.*;
import java.io.*;

public class atcoder_ABC132_F
{
    public static void main ( String[] args ) throws java.lang.Exception {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        int mod = 1000000007;
        int coef[] = new int[n];
        for(int i = 1; i <= n/2; i++) {
            coef[i] = (n/i) - (n/(i+1));
        }
        coef[n/2] = 1;
        int nn = coef.length;
        int dp[][] = new int[k+1][nn];
        dp[0][0] = 1;
        for(int i = 1; i <= k; i++) {
            int tmp = 0;
            for(int j = nn-1; j >= 0; j--) {
                tmp += dp[i-1][(nn-1)-j];
                tmp %= mod;
                dp[i][j] = coef[j] * tmp;
                dp[i][j] %= mod;
            }
        }
        int sum = 0;
        for(int i = 0; i < nn; i++) {
            sum += dp[k][i];
            sum %= mod;
        }
        System.out.println(sum);
    }
}

File: output/CodeGen/avatar/Java/Python/atcoder_ABC132_F.py
def main(args):
    mod = 1e9 + 7
    sc = Scanner(sys.stdin)
    n = sc.nextInt()
    k = sc.nextInt()
    sc.close()
    len = [1]
    while len[-1] <= n:
        len.append(len[-1] + len[-2])
    q = len[-1]
    dp = [0] * (k * (q + 1))
    for j in range(1, q + 1):
        dp[j] = len[j - 1] + dp[j - 1]
    for i in range(1, k):
        for j in range(1, q + 1):
            dp[i * (q + 1) + j] = dp[i * (q + 1) + j - 1] + dp[(i - 1) * (q + 1) + q - j + 1] * len[j - 1]
            dp[i * (q + 1) + j] %= mod
    print(dp[k * (q + 1) - 1])